PixelHunter1995/SceneParser.cs

```diff
@@ -0,0 +1,39 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+using System.Xml;
+
+namespace PixelHunter1995
+{
+    class SceneParser
+    {
+        public static void ParseSceneXml(String sceneXmlPath)
```

Comment:
Perhaps I'm thinking ahead. But I imagine this function should just return an object "Scene" containing all the data. 

And kinda related, should we ever write the xml when the scene gets deloaded, and it's been modified? Or should we just keep these objects in memory? My hunch is we shouldn't write it, but I guess it might help debugging.

Reply:
Yes, this function returning a scene object seems right.

I don't think we should ever modify the scene XML. Keeping objects in memory seems more correct. I'm not even sure if we should modify the scene objects. In Hundens Vagga I keep a "world state" object, containing info about what has happened in the world (e.g. if a door is unlocked). Since some state is not really related to a certain scene (e.g. "the electricity is on"), I think that will be easier. Then, scenes can contain data about how it will look when the world state has been altered.

But I haven't really thought this through, more than when writing Hundens Vagga. Perhaps we can come up with a better way to do it for this game, which after all will be more complex.

Saving the game in Hundens Vagga simply means writing the inventory items, the world state and the current room to a file. An example save file looks like this:
```
letter,mainkey,flashlight
SUBTITLES_ON:False,st_frozen_melted:False,st_shovel_taken:False,st_crawlspace_open:False,st_crawlspace_illuminated:False,st_water_running:False,st_window_open:False,st_handle_taken:False,st_maindoor_open:False,st_doormat_moved:True,st_mainkey_taken:True,st_flashlight_taken:True,st_typewriter_room_open:False,st_maps_open:False,st_notebook_taken:False,st_toykey_taken:False,st_wrench_taken:False,st_electricity_on:False,st_saucepan_taken:False,st_drawers_open:False,st_fuse_taken:False,st_microwave_open:False,st_typewriterkey_taken:False,st_toys_open:False,st_treasure_dug:False,st_musicbox_taken:False
car
```